### PR TITLE
[Bugfix][Wan2.2] Apply UniPC flow shift once

### PR DESCRIPTION
**Pull request by HAAZZZEEEE**
## Purpose

Align Wan2.2 UniPC schedules with native Wan for T2V, I2V, VACE, and S2V. Constructing UniPC with the requested flow shift changes the training sigma endpoints; inference then interpolates between those endpoints and applies the shift again. For T2V/I2V at 40 steps and shift 12, this changes 6 integer timesteps and the first sigma from approximately 0.9999166 to 0.9999930.

All four pipeline paths now construct UniPC with `shift=1.0` and pass the requested shift explicitly to `set_timesteps`. S2V retains its default shift of 3.0. The T2V/I2V path no longer calls `set_shift`, so it does not mutate Diffusers' FrozenDict or make attribute and dictionary config reads disagree. Euler and fixed DMD scheduling remain unchanged.

Compatibility: identical parameters and seeds can produce different outputs from older versions. Pin the code revision when reproducing historical results, and use the same scheduler revision for BF16/quantized comparisons. No video-quality or performance improvement is claimed.

### Reference and scope

The golden fixture is reproduced from [official Wan2.2 UniPC](https://github.com/Wan-Video/Wan2.2/blob/42bf4cfaa384bc21833865abc2f9e6c0e67233dc/wan/utils/fm_solvers_unipc.py), SHA-256 `0dec8c7ed17f6f2049275c6848113314da6ccec1c8db5bdc89df43c05c6038d9`. [Wan's T2V call site](https://github.com/Wan-Video/Wan2.2/blob/42bf4cfaa384bc21833865abc2f9e6c0e67233dc/wan/text2video.py#L309-L314) initializes with shift 1 and passes the requested shift to `set_timesteps`.

This scheduler fix was extracted from #7210 and has no quantization dependency. The shared UniPC implementation is unchanged. #7053 concerns scalar-update synchronization rather than endpoint construction.

## Test Plan

Prerequisites: normal vLLM-Omni development/test dependencies, including compatible vLLM, PyTorch, Diffusers, and pytest. These tests run on CPU without model weights.

```bash
python -m pytest -q tests/diffusion/models/wan2_2/test_wan22_scheduler.py tests/diffusion/models/wan2_2/test_wan_euler_scheduler_dmd.py tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py tests/diffusion/models/wan2_2/test_wan22_s2v_pipeline.py tests/diffusion/models/wan2_2/test_wan22_i2v_pipeline.py tests/diffusion/models/wan2_2/test_wan22_vace_pipeline.py

# CI-like selection of the same CPU regressions:
python -m pytest -q tests/diffusion/models/wan2_2/test_wan22_scheduler.py tests/diffusion/models/wan2_2/test_wan_euler_scheduler_dmd.py tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py tests/diffusion/models/wan2_2/test_wan22_s2v_pipeline.py tests/diffusion/models/wan2_2/test_wan22_i2v_pipeline.py tests/diffusion/models/wan2_2/test_wan22_vace_pipeline.py -m 'core_model and cpu' --run-level core_model
```

## Test Result

Tested revision: `e5c239be0ac0818bf6515fbc56cdfb54402a8ce6`.

- **86 passed, 18 warnings in 2.81s**, normal repository conftests and real pipeline imports. Warnings concern torch JIT deprecation and CPU autocast in mocked-model tests.
- **6 failed / 80 passed** with the same tests and only I2V/VACE production files restored from `1a8aba23`. All six failures are sigma mismatches at shifts 3/5/12 with 50 steps. Euler cases pass in both versions.
- I2V/VACE tests exercise actual forward scheduling with real schedulers and compare every UniPC sigma and integer timestep against independent arithmetic. Heavy model components remain mocked. Existing native-golden, T2V, S2V, and Euler/DMD coverage remains in the six-file suite.
- Linux x86-64 CPU container, 4 CPUs / 8 GiB, no accelerator devices: Python 3.12.13, torch 2.10.0+cpu, NumPy 2.5.2, Diffusers 0.40.0, vLLM 0.29.0+empty, transformers 5.14.1, pytest 9.1.1. Same dedicated CPU environment as the previous review run, hardware plugins disabled.
- Applicable changed-file pre-commit hooks and git diff --check passed. Actionlint and Markdownlint skipped because no workflow or Markdown source files change. Tested source archive matches the committed working tree.
- GPU/NPU inference, video quality, and performance have not been validated for this revision.

### Previous CI failures investigated

Both failed checks on `1a8aba23` occurred before tests ran:

- [Intel build 8302](https://buildkite.com/vllm/vllm-omni-intel-ci/builds/8302): checkout failed with `No space left on device`, exit 128.
- [NPU build 6951](https://buildkite.com/vllm/vllm-omni-npu-ci/builds/6951): Wan I2V pod remained Pending for 30m1s, unschedulable due to insufficient CPU, Ascend, or attachable-volume resources. The job never started.

These infrastructure failures are separate from the I2V/VACE shift regression fixed here. Latest-revision CI remains the acceptance gate.

AI assistance: Codex implemented the fixes, inspected CI failures, ran CPU regressions/static checks, and drafted this description. Automated checks do not constitute human review.

### CODE REVIEW

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py [hunk 1]
```diff
@@ -59,11 +59,15 @@
 
 def build_wan_scheduler(sample_solver: str, flow_shift: float) -> Any:
     if sample_solver == "unipc":
-        return FlowUniPCMultistepScheduler(
+        # Keep native Wan's unshifted training endpoints; set_timesteps applies
+        # the requested shift to the interpolated inference sigmas.
+        scheduler = FlowUniPCMultistepScheduler(
             num_train_timesteps=1000,
-            shift=flow_shift,
+            shift=1.0,
```
Comment (david6666666) [inline vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:66]: Does S2V need the same fix? `pipeline_wan2_2_s2v.py:549` still constructs with `shift=flow_shift`, then calls `set_timesteps(num_steps, device=device)` at :1345 with no shift, so it double-applies too (default 3.0). Native S2V uses `shift=1` plus `set_timesteps(..., shift=shift)`.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py [hunk 2]
```diff
@@ -59,11 +59,15 @@
...
             prediction_type="flow_prediction",
         )
+        scheduler.set_shift(flow_shift)
```
Comment (david6666666) [inline vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py]: `set_shift` writes `self.config.shift`, and `config` is a diffusers `FrozenDict`. The write only lands because `FrozenDict.__setattr__` tests `hasattr(self, "__frozen")` while the real flag is name-mangled to `_FrozenDict__frozen`, so the freeze never fires. Afterwards `config["shift"]` still reads 1.0 while `config.shift` reads the runtime value. Other UniPC call sites here (magi2, dreamzero) keep `shift=1.0` and pass `shift=` to `set_timesteps` instead — worth doing that?

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py (hunk 1 above)
Comment (HAAZZZEEEE) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:66]: Fixed in 1a8aba23. S2V now constructs UniPC with shift=1.0, retains its configured/default (3.0) flow shift, and passes it explicitly to set_timesteps for each clip. Added constructor coverage for default/custom shifts and actual S2V forward scheduling checks against independent five-step sigma arithmetic at shifts 3 and 12. The four-file CPU suite passes 51 tests; restoring the previous two pipeline files makes 15 tests fail. Model components are mocked in pipeline unit tests; no S2V video-quality or accelerator inference claim.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py (hunk 2 above)
Comment (HAAZZZEEEE) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py]: Agreed; fixed in 1a8aba23. Removed set_shift from the Wan factory. T2V/I2V now supplies shift explicitly to set_timesteps only for UniPC; Euler keeps its existing call. S2V uses explicit runtime shift too. Tests assert config.shift == config["shift"] == 1.0 after scheduling and exercise actual T2V forward calls with request shifts 3 -> 12 -> 5. The previous revision reproduces the FrozenDict attribute/key disagreement; the updated four-file suite passes 51 tests.

**Comment by HAAZZZEEEE** [comment]
Self-review updated for e5c239be (Codex-assisted).

- Checked all Wan scheduler factory callers, including the separate I2V path and inherited VACE path. T2V, I2V, VACE and S2V now pass the runtime shift explicitly to UniPC, with unshifted initialization endpoints. Config attribute/key reads remain consistent, and Euler/DMD behavior is preserved.
- **The latest code passes all 86 CPU tests across six related test files.** This includes I2V and VACE tests using real schedulers to verify the sigma values and timesteps for all 50 sampling steps.
- To confirm the new tests catch the bug, I kept those tests and temporarily restored the old I2V/VACE code from `1a8aba23` in a separate test copy. As expected, six tests failed: I2V and VACE each tested with shifts 3, 5 and 12. The old code ignored the requested shift and used 1 instead. **These are deliberately reproduced failures on the old code, not failures on the latest PR revision.** All six pass with the fix.
- Applicable changed-file pre-commit hooks and git diff --check passed. Actionlint and Markdownlint were skipped since no workflow or Markdown source files changed.

Pipeline unit tests still mock the heavy model components. GPU/NPU inference, video quality and performance remain unvalidated for this revision. The PR description includes the test environment and reproduction commands. This supersedes the earlier 51-test self-review, which missed the I2V/VACE callers; automated checks do not constitute human review.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py [hunk 3]
```diff
@@ -59,9 +59,11 @@
 
 def build_wan_scheduler(sample_solver: str, flow_shift: float) -> Any:
     if sample_solver == "unipc":
+        # Keep native Wan's unshifted training endpoints; set_timesteps applies
+        # the requested shift to the interpolated inference sigmas.
         return FlowUniPCMultistepScheduler(
             num_train_timesteps=1000,
-            shift=flow_shift,
+            shift=1.0,
```
Comment (david6666666) [inline vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:66]: Now that the factory always builds with `shift=1.0`, each caller has to pass the shift to `set_timesteps`, but only T2V was updated. `pipeline_wan2_2_i2v.py:632` and `pipeline_wan2_2_vace.py:693` still call `set_timesteps(num_steps, device=device)`, so `shift` falls back to `config.shift == 1.0` and `flow_shift` becomes a no-op for I2V/VACE (defaults 5.0 / 3.0). At flow_shift=5.0 and 50 steps, I2V's schedule goes from 10/50 integer timesteps off native to 49/50, with sigma_max stuck at 0.999 instead of 0.9998. I2V is what the `diffusion_wan22_accuracy` lane measures, and its unit tests inject `StubScheduler`, so nothing currently catches this. These two need the same `unipc` guard as line 779.

Comment (HAAZZZEEEE) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:66]: Thanks for catching this, I missed the separate I2V and VACE paths.
I'll fix both call sites, check the other callers too, and add tests with the real scheduler. Will keep Euler unchanged.

Comment (HAAZZZEEEE) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:66]: Fixed in e5c239be, both I2V and VACE now pass the runtime shift for UniPC. Checked all factory callers and the inherited VACE path too. Forward tests now use real schedulers, covering shifts 3/5/12 at 50 steps and Euler. Restoring the previous I2V/VACE files gives 6 sigma-mismatch failures, the updated six-file suite passes all 86 tests.

**Review by david6666666 — APPROVED** [review]
Approved.

The shift is now applied exactly once across every Wan2.2 path, matching native Wan: T2V and I2V pass the per-request `flow_shift` to `set_timesteps` only for UniPC, VACE does the same with its init-time `_flow_shift` (default 3.0), and S2V constructs with `shift=1.0` then passes `self._flow_shift` at scheduling. Euler keeps its existing `_shift` handling, and `FlowUniPCMultistepScheduler.set_shift` is no longer called, so the `FrozenDict` shadowing concern is gone.

The forward tests now use the real scheduler and assert independent sigma/timestep arithmetic for unipc at shifts 3/5/12 and for Euler at 50 steps. I checked the schedule is bit-identical to the native shift-12 fixture and to shift-once arithmetic for T2V (5/12), I2V (5), S2V (3), and VACE (3).

CI: DCO, build (3.11/3.12), pre-commit, readthedocs, and the CUDA/AMD/NPU Buildkite lanes all pass. The Intel lane is red, but it fails on base and on main tip too, so it's unrelated to this change.
